packages/cli/src/code-format.ts

```diff
@@ -0,0 +1,194 @@
+import { writeFileSync } from 'fs';
+
+const { join } = nodeFs;
+
+const argv = yargs
+    .usage('$0 [options]')
+    .option('target', {
+        type: 'string',
+        description: 'file or directory to format',
+        alias: 'T',
+        default: process.cwd(),
+        defaultDescription: 'current working directory',
+    })
+    .option('endWithNewline', {
+        type: 'boolean',
+        description: 'End output with newline',
+        alias: 'n',
+        default: false,
+    })
+    .option('indentEmptyLines', {
+        type: 'boolean',
+        description: 'Keep indentation on empty lines',
+        alias: 'E',
+        default: false,
+    })
+    .option('indentSize', {
+        type: 'number',
+        description: 'Indentation size',
+        alias: 's',
+        default: 4,
+    })
+    .option('indentWithTabs', {
+        type: 'boolean',
+        description: 'Indent with tabs, overrides -s and -c',
+        alias: 't',
+        default: false,
+    })
+    .option('maxPerserveNewlines', {
+        type: 'number',
+        description: 'Maximum number of line-breaks to be preserved in one chunk',
+        alias: 'm',
+        default: 1,
+    })
+    .option('newlineBetweenRules', {
+        type: 'boolean',
+        description: 'Add a newline between CSS rules',
+        alias: 'N',
+        default: true,
+    })
+    .option('perserveNewlines', {
+        type: 'boolean',
+        description: 'Preserve existing line-breaks',
+        alias: 'p',
+        default: true,
+    })
+    .option('selectorSeparatorNewline', {
+        type: 'boolean',
+        description: 'Add a newline between multiple selectors',
+        alias: 'L',
+        default: true,
+    })
+    .option('debug', {
+        type: 'boolean',
+        description: 'Enable explicit debug log (overrides --silent)',
+        alias: 'D',
+        default: false,
+    })
+    .option('silent', {
+        type: 'boolean',
+        description: 'Will not print any messages to the log',
+        alias: 'S',
+        default: false,
+    })
+    .option('require', {
+        type: 'array',
+        description: 'require hooks',
+        alias: 'r',
+        default: [] as string[],
+    })
+
+    .alias('h', 'help')
+    .alias('v', 'version')
+    .help()
+    .strict()
+    .wrap(yargs.terminalWidth())
+    .parseSync();
+
+const {
+    debug,
+    endWithNewline,
+    indentEmptyLines,
+    indentSize,
+    indentWithTabs,
+    maxPerserveNewlines,
+    newlineBetweenRules,
+    perserveNewlines,
+    require: requires,
+    selectorSeparatorNewline,
+    target,
+    silent,
+} = argv;
+
+const log = createLogger('[Stylable code formatter]', true);
+
+if (debug) {
+    log('[Arguments]', argv);
+}
+
+// execute all require hooks before running the CLI build
+for (const request of requires) {
+    if (request) {
+        require(request);
+    }
+}
+
+function readDirectoryDeep(dirPath: string, fileSuffixFilter = '.st.css') {
```

Comment:
we already have this function directoryDeepChildren can we reuse?

Reply:
It's async which kinda makes everything just a little bit trickier. Think we can avoid it in this case.